lib/tpm2_policy.c

```diff
@@ -200,6 +200,57 @@ bool tpm2_policy_build_pcr(TSS2_SYS_CONTEXT *sapi_context,
     return result;
 }
 
+bool tpm2_policy_build_policyAuthorize(TSS2_SYS_CONTEXT *sapi_context,
+    tpm2_session *policy_session, const char *to_authorize_policy_file,
+    const char *to_authorize_policy_qualifier_file,
+    const char *verifying_public_key_name_file,
+    const char *verification_ticket_file) {
+
+    TPM2B_DIGEST approvedPolicy = {
+        .size = TPM2_SHA256_DIGEST_SIZE
```

Comment:
Your hardcoding the size of sha256, shouldn't his be algorithm flexible? Can we just use the size of the file to figure it out or should the user specify the algorithm. SHA isn't the only digester on the block, so we should probably have user specified.